packages/lib/responses.ts

```diff
@@ -39,3 +39,21 @@ export const getQuestionResponseMapping = (
 
   return questionResponseMapping;
 };
+
+export const processResponseData = (
+  responseData: string | number | string[] | Record<string, string>
+): string => {
+  if (!responseData) return "";
+  if (typeof responseData === "string") {
+    return responseData;
+  } else if (typeof responseData === "number") {
+    return responseData.toString();
+  } else if (Array.isArray(responseData)) {
+    return responseData.join("; ");
+  } else {
+    const formattedString = Object.entries(responseData)
+      .map(([key, value]) => `${key}: ${value}`)
+      .join("\n");
+    return formattedString;
+  }
+};
```

Comment:
Can we convert this to a `switch`? Since it has so many conditional statements & returns based on that, looks like a better option?